HRIS.Services/Services/DashboardService.cs

```diff
@@ -0,0 +1,247 @@
+﻿using HRIS.Models;
+using HRIS.Services.Interfaces;
+using RR.UnitOfWork;
+using RR.UnitOfWork.Entities.HRIS;
+
+namespace HRIS.Services.Services
+{
+    public class DashboardService : IDashboardService
+    {
+        private readonly IUnitOfWork _db;
+
+        public DashboardService(IUnitOfWork db)
+        {
+            _db = db;
+        }
+        public async Task<ChurnRateDataCardDto> CalculateEmployeeChurnRate()
+        {
+            var today = DateTime.Today;
+            var twelveMonthsAgo = today.AddMonths(-12);
+
+            var employeeData = await _db.Employee
+                .GetAll();
+
+            var employeeStartOfPeriod = employeeData
+                .Where(e => e.EngagementDate < twelveMonthsAgo && e.Active == true)
+                .ToList();
+
+            var terminatedEmployeesEndOfPeriod = employeeData
+                .Where(e => e.Active == false && e.TerminationDate >= twelveMonthsAgo)
+                .ToList();
+
+            var employeeStartOfPreviousPeriod = employeeData
+                 .Where(e => e.EngagementDate < twelveMonthsAgo.AddMonths(-12) && e.Active == true)
+                .ToList();
+
+            var terminatedEmployeesEndOfPreviousPeriod = employeeData
+                .Where(e => e.Active == false && e.TerminationDate >= twelveMonthsAgo.AddMonths(-12))
+                .ToList();
+
+            int GetCount(List<Employee> data, int employeeType) => data.Count(x => x.EmployeeTypeId == employeeType);
+
+            var churnRate = CalculateChurnRate(employeeStartOfPeriod.Count, terminatedEmployeesEndOfPeriod.Count);
+         
+            var previousChurnRate = CalculateChurnRate(employeeStartOfPreviousPeriod.Count, terminatedEmployeesEndOfPreviousPeriod.Count);
+
+            var averageChurnRate = (previousChurnRate + churnRate) / 2;
+            var percentageDifference = averageChurnRate != 0
+                ? ((previousChurnRate - churnRate) / averageChurnRate) * 100
+                : 0;
+
+            bool isIncrease = percentageDifference > 0;
```

Comment:
```suggestion
            var isIncrease = percentageDifference > 0;
```

Just for consistency use `var` where possible

Reply:
no problem